### ISSUE TITLE
New dissolution imputation error in netdx

### ISSUE THREAD

**Issue by thednainus** [statnet/EpiModel#529]
Hello,

I have recently updated EpiModel in my computer and I noticed that I get the following warning when running `netdx` .
I updated to 2.0.5 (from CRAN) and 2.0.6 (from GitHub) and in both versions I get the same warning message.

```
Warning message:
In x$coef.diss$duration^2 * dgeom(2:(nsteps + 1), 1/x$coef.diss$duration) :
  longer object length is not a multiple of shorter object length
```

I don't get this warning when using Epimodel version 2.0.4 (which I have in the cluster).
Should I be worried about this warning?

Thanks!
Fabricia.

**Comment by smjenness** [comment]
@thednainus : we recently updated our dissolution model diagnostics. Can you provide a minimally reproducible example so we can test/fix?

cc: @sgoodreau

**Comment by thednainus** [comment]
Sure, you can find an example here: https://github.com/thednainus/HIVepisim/blob/master/Analyses/MSM_Demog_Migration_example.R

I don't think you need to download the R package I am developing to run this script because this is only the fit.

Cheers,
Fabricia.

**Comment by sgoodreau** [comment]
Hi Fabricia,

Thank you very much for the clear example - I was able to run it without needing your extra package. It looks like you have a heterogeneous dissolution model, something which works fine in EpiModel per se, but which we have never developed good diagnostics for.  In the prior version of EpiModel, this issue would only appear when you went to print or plot the diagnostics, when it would give an informative warning that the dissolution diagnostics are not well-developed.  In the re-working of the method of homogenous dissolution diagnostics that I did for this version, I didn't account for the fact that it would now generate the non-informative (and confusing) warning that you encountered during the previous step, when `netdx` itself is run.  I'll work to fix that for the next version. In either case, it doesn't affect your actual model since it's just for diagnostics.

Thanks!
